pkg/parser/lexer.go

```diff
@@ -187,7 +187,7 @@ func (l *lexer) lexString(lval *extendedSyntaxSymType) int {
 	for r := l.next(); r != '"' && r != '\''; r = l.next() {
 		if r == eof {
 			l.previous = l.start
-			// return error position
+			l.err = fmt.Errorf("unmatched quotation")
```

Comment:
Looks like the parser is depending on the `.err` field on the lexer to determine when to stop. I will go through related documentation on this subject.